### PR TITLE
Removed dependence on c11 features (auto type)

### PR DESCRIPTION
**Pull request by vo1stv**
Not sure what your motivation for using c11 features is.
I believe less is more; the changes in this branch allow the code to be built without c11 features.
I have built this on Ubuntu, but I haven't tested on other flavours of Linux or on Windows or Mac, so if your motivation for c11 has to do with cross-platform compatibility, you can reject this pull request.

### CODE REVIEW

**Comment by dahn-d** [comment]
Why and how may `auto` hurt?

**Comment by aardappel** [comment]
I personally prefer code that uses `auto` everywhere, it is more readable to me, less type clutter. But this code mostly predates my switch to C++11, so I don't mind keeping it compatible with older C++ for now. Just a few corrections.

File: src/Makefile [hunk 1]
```diff
@@ -6,7 +6,7 @@ WX_CXXFLAGS=	`$(WX_CONFIG) --cxxflags`
 WX_LIBS=	`$(WX_CONFIG) --libs aui adv core xml net`
 PACKAGE_VERSION?=	__DATE__
 CXX?=	c++
-CXXFLAGS+=	-DPACKAGE_VERSION=$(PACKAGE_VERSION) -std=c++11 $(WX_CXXFLAGS)
```
Comment (aardappel) [inline src/Makefile]: Let's keep this in. TreeSheets is primarily developed on Windows, and I don't want my linux builds to break when a commit accidentally uses auto.

File: src/document.h [hunk 2]
```diff
@@ -1225,18 +1225,18 @@ struct Document
             case A_CUSTKEY:
             {
                 wxArrayString strs;
-                auto &ms = sys->frame->menustrings;
-                for (auto it = ms.begin(); it != ms.end(); ++it) strs.push_back(it->first);
+                MyFrame::MenuString & ms = sys->frame->menustrings;
```
Comment (aardappel) [inline src/document.h]: Please use the existing style of the surrounding code, meaning no space after &

File: src/myframe.h [hunk 3]
```diff
@@ -1,5 +1,7 @@
 struct MyFrame : wxFrame
 {
+    typedef std::vector<std::pair<wxString, wxString> > MenuString;
+    typedef std::vector<std::pair<wxString, wxString> >::iterator MenuStringIterator;
```
Comment (aardappel) [inline src/myframe.h]: This is why you use auto :)

Comment (vo1stv) [inline src/myframe.h]: Agreed auto works nicely for iterators.

File: src/Makefile (hunk 1 above)
Comment (vo1stv) [inline src/Makefile]: Not sure what you mean here.

**Comment by vo1stv** [comment]
I'm a bit new to the GitHub PR workflow.  I think I've addressed your concerns (waiting on clarification for Makefile note)

**Comment by aardappel** [comment]
Looks good now, thanks.

**Pull request merged**